src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -469,25 +469,32 @@ private bool ReportConflictWithLocal(Symbol local, Symbol newSymbol, string name
                 return true;
             }
 
-            if (newSymbolKind == SymbolKind.Local || newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Method || newSymbolKind == SymbolKind.TypeParameter)
```

Comment:
>if (newSymbolKind == SymbolKind.Local || newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Method || newSymbolKind == SymbolKind.TypeParameter) [](start = 12, length = 162)

Instead of making changes to this function would it be simpler to make changes to ```ValidateNameConflictsInScope``` method to get out of the loop once we are done with a body of a local function and the language version is right?